hkmc2/shared/src/main/scala/hkmc2/syntax/Tree.scala

```diff
@@ -73,13 +82,47 @@ enum Tree extends AutoLocated:
     case InfixApp(lhs, _, rhs) => Ls(lhs, rhs)
     case TermDef(k, symName, alphaName, rhs) => symName.toList ++ alphaName ++ rhs
     case New(body) => body :: Nil
+    case If(split) => split :: Nil
     case IfElse(cond, alt) => cond :: alt :: Nil
     case Case(bs) => Ls(bs)
     case Region(name, body) => name :: body :: Nil
     case RegRef(reg, value) => reg :: value :: Nil
     case Effectful(eff, body) => eff :: body :: Nil
   
-  def describe: Str = ??? // TODO
+  def describe: Str = this match
+    case Empty() => "empty"
+    case Error() => "error"
+    case Ident(name) => name
+    case IntLit(value) => value.toString
+    case DecLit(value) => value.toString
+    case StrLit(value) => value.escaped
+    case UnitLit(value) => if value then "undefined" else "null"
+    case BoolLit(value) => value.toString
+    case Block(stmts) => stmts.map(_.describe).mkString("{ ", "; ", " }")
+    case Let(lhs, rhs, body) => s"let $lhs = $rhs in $body"
+    case TermDef(k, symName, alphaName, rhs) =>
+      val name = symName.orElse(alphaName).fold("")(n => s" ${n.describe}")
+      val sign = rhs.fold("")(s => s" : ${s.describe}")
+      s"${k.str}$name$sign"
```

Comment:
This is crazy. I assume this is Copilot of ChatGPT junk. The goal of `describe` is simply to name what tree we're looking at using a word or two.

Reply:
You're absolutely right. I didn't look into this case. I will remove it soon.